src/services/EventBus.ts

```diff
@@ -30,4 +35,47 @@ EventBus.once = function<Key extends keyof Events>(type: Key, handler: GenericEv
 		this.off(type, fn)
 	}
 	this.on(type, fn)
+
+	// Store reference to the original handler to be able to remove it later
+	if (!EventBus._onceHandlersMap.has(type)) {
+		EventBus._onceHandlersMap.set(type, new Map())
+	}
+	EventBus._onceHandlersMap.get(type)!.set(handler, fn)
+}
+
+/**
+ * OVERRIDING OF ORIGINAL MITT FUNCTION
+ * Remove an event handler for the given type.
+ * If `handler` is omitted, all handlers of the given type are removed.
+ * @param type Type of event to unregister `handler` from (`'*'` to remove a wildcard handler)
+ * @param [handler] Handler function to remove
+ */
+EventBus.off = function<Key extends keyof Events>(type: Key, handler?: GenericEventHandler) {
+	const handlers: Array<GenericEventHandler> | undefined = EventBus.all!.get(type)
+	const onceHandlers: Map<GenericEventHandler, GenericEventHandler> | undefined = EventBus._onceHandlersMap!.get(type)
+
+	if (handlers) {
+		if (handler) {
+			handlers.splice(handlers.indexOf(handler) >>> 0, 1)
+		} else {
+			EventBus.all!.set(type, [])
+		}
+	}
+
+	if (handlers && onceHandlers) {
+		if (handler) {
+			for (const [_handler, fn] of onceHandlers) {
+				if (handler === fn) {
+					// Event was received once and removed by original code, clearing reference only
+					onceHandlers.delete(_handler)
+				} else if (_handler === handler) {
+					// Event was not received yet, removing the fn and reference
+					handlers.splice(handlers.indexOf(fn) >>> 0, 1)
+					onceHandlers.delete(_handler)
+				}
+			}
+		} else {
+			EventBus._onceHandlersMap.set(type, new Map())
+		}
+	}
```

Comment:
We don't need to completely re-implement it duplicating all the `mitt` logic. We can extend it, reusing the original method.

```suggestion
const off = EventBus.off.bind(EventBus)
EventBus.off = function<Key extends keyof Events>(type: Key, handler?: GenericEventHandler) {
  off(type, handler)
  const onceHandler = handler && onceHandlers.get(type)?.get(handler)
  if (onceHandler) {
    onceHandlers.get(type).remove(handler)
    off(type, onceHandler)
  }
}
```

Reply:
It didn't pass these tests, so adjusted in fixups step by step: